README.md

```diff
@@ -132,6 +132,50 @@ until the constructor for `config.Provider` can be fully satisfied.
 If resolution is not possible (for instance, if one of the required dependencies
 lacks a constructor and doesn't appear in the graph), `dig` returns an error.
 
+## Invoke
+
+The `Invoke` API executes the provided function and stores returned results in the dig container.
+
+`Invoke` looks through the dig graph and resolves all the constructor parameters for execution.
+At the end of method invokation, `Invoke` inserts return results except the `error` object, into the
+graph for further use.
+To use `Invoke`, make sure your function meets the following criteria:
+
+1. Arguments must be pointers, maps, slices, or arrays
+2. Arguments must be provided to dig
+3. (Optional) Function has return parameters
+
+For example, `Invoke` scenarios:
```

Comment:
These are example `Invoke` scenarios? Can you introduce what's going on a bit more? Like, "For example, see how `Invoke` works in…".